app/controllers/nonsipp/BasicDetailsCheckYourAnswersController.scala

```diff
@@ -0,0 +1,291 @@
+import utils.ListUtils.ListOps
+import viewmodels.DisplayMessage._
+import viewmodels.implicits._
+import viewmodels.models._
+import views.html.CheckYourAnswersView
+
+import java.time.LocalDate
+import javax.inject.{Inject, Named}
+
+class BasicDetailsCheckYourAnswersController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: CheckYourAnswersView
+) extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
+    taxYearOrAccountingPeriods(srn) match {
+      case Some(periods) =>
+        (
+          for {
+            schemeMemberNumbers <- requiredPage(HowManyMembersPage(srn, request.pensionSchemeId))
+            activeBankAccount <- requiredPage(ActiveBankAccountPage(srn))
+            whyNoBankAccount = request.userAnswers.get(WhyNoBankAccountPage(srn))
+            userName <- loggedInUserNameOrRedirect
+          } yield Ok(
+            view(
+              viewModel(
+                srn,
+                mode,
+                schemeMemberNumbers,
+                activeBankAccount,
+                whyNoBankAccount,
+                periods,
+                userName,
+                request.schemeDetails,
+                request.pensionSchemeId.value
+              )
+            )
+          )
+        ).merge
+      case _ => Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+    }
+  }
+
+  def onSubmit(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
+    Redirect(navigator.nextPage(BasicDetailsCheckYourAnswersPage(srn), mode, request.userAnswers))
+  }
+
+  private def taxYearOrAccountingPeriods(
+    srn: Srn
+  )(implicit request: DataRequest[_]): Option[Either[DateRange, NonEmptyList[(DateRange, Max3)]]] =
+    request.userAnswers.list(AccountingPeriods(srn)) match {
+      case Nil => request.userAnswers.get(WhichTaxYearPage(srn)).map(Left(_))
+      case head :: rest =>
+        NonEmptyList
+          .of(head, rest: _*)
+          .zipWithIndex
+          .traverse {
+            case (date, index) => refineV[OneToThree](index + 1).toOption.map(refined => date -> refined)
+          }
+          .map(Right(_))
+    }
+
+  private def loggedInUserNameOrRedirect(implicit request: DataRequest[_]): Either[Result, String] =
+    request.minimalDetails.individualDetails match {
+      case Some(individual) => Right(individual.fullName)
+      case None =>
+        request.minimalDetails.organisationName match {
+          case Some(orgName) => Right(orgName)
+          case None => Left(Redirect(controllers.routes.JourneyRecoveryController.onPageLoad()))
+        }
+    }
+
+  private def requiredPage[A: Reads](page: Gettable[A])(implicit request: DataRequest[_]): Either[Result, A] =
```

Comment:
This would be useful for other controllers, think about putting it somewhere shared

Reply:
You did something similar but as an action but it only allows a single page to be required, i'll look at refactoring this out in the future and adapting your code to take multiple pages